src/modules/roc_pipeline/converter_source.cpp

```diff
@@ -68,6 +68,17 @@ ConverterSource::ConverterSource(const ConverterConfig& config,
         areader = pipeline_poisoner_.get();
     }
 
+    if (config.profiling) {
+        profiler_.reset(new (allocator) audio::ProfilingReader(
```

Comment:
Since profiler is now on top of poisoner, we should also move `profiler_` field after `pipeline_poisoner_` field here: https://github.com/roc-project/roc/blob/7a802f6e966e0702a2fba6c6b4b3adef0e5fbf1b/src/modules/roc_pipeline/converter_sink.h

We need it to ensure that poisoner is not destroyed before profiler, because profiler object uses poisoner object (although in current profiler implementation doesn't use nested reader, it may change in future).